runtime_tests/SamV71_HWAS/work/controller/C/src/controller.c

```diff
@@ -0,0 +1,625 @@
+    REGISTER_VALUE_EQUAL(&SPI0_MR, EXPECTED_SPI0_MR);
+    REGISTER_VALUE_EQUAL(&SPI0_CSR0, EXPECTED_SPI0_CSR0);
+}
+
+static inline void Test_read_write_buffer(void)
+{
+    asn1SccByteBuffer buffer = {.nCount = NUMBER_OF_REGISTERS_TO_CHECK};
+    asn1SccByteBuffer actualBuffer = {.nCount = NUMBER_OF_REGISTERS_TO_CHECK};
+
+    /// Enable and reset SPI0
+    Init_spi_setup_pmc_and_pio();
+    Spi_reset();
+    /// Read SPI0 config after reset
+    controller_RI_RawMemoryAccess_ReadBuffer_Pi(&SPI0_CR, &buffer);
+    /// Check if SPI0 config is as expected
+    BUFFER_VALUE_EQUAL(&buffer, EXPECTED_SPI0_REGISTERS_AFTER_RESET);
+
+    /// Enable SPI0
+    Spi_enable();
+    /// Read SPI0 register value after enabling it
+    controller_RI_RawMemoryAccess_ReadBuffer_Pi(&SPI0_CR, &buffer);
+
+    /// Prepare SPI0 configuration
+    buffer.arr[SPI0_CR_REGISTER_INDEX] = SETUP_SPI0_CR;
+    buffer.arr[SPI0_MR_REGISTER_INDEX] = SETUP_SPI0_MR;
+    buffer.arr[SPI0_CSR0_REGISTER_INDEX] = SETUP_SPI0_CSR0;
+
+    /// Configure SPI0
+    controller_RI_RawMemoryAccess_WriteBuffer_Pi(&SPI0_CR, &buffer);
+
+    /// Check if SPI0 was properly configured
+    buffer.arr[SPI0_CR_REGISTER_INDEX] = EXPECTED_SPI0_CR;
+    buffer.arr[SPI0_MR_REGISTER_INDEX] = EXPECTED_SPI0_MR;
+    buffer.arr[SPI0_RDR_REGISTER_INDEX] = EXPECTED_SPI0_RDR;
+    buffer.arr[SPI0_SR_REGISTER_INDEX] = EXPECTED_SPI0_SR;
+    buffer.arr[SPI0_CSR0_REGISTER_INDEX] = EXPECTED_SPI0_CSR0;
+
+    controller_RI_RawMemoryAccess_ReadBuffer_Pi(&SPI0_CR, &actualBuffer);
+    BUFFER_VALUE_EQUAL(&actualBuffer, buffer.arr);
+}
+
+static inline void Test_read_modify_write_success(void) {
+    asn1SccWord actualData = 0x00;
+    const asn1SccSourceAddress testDataAddress =
+        (const asn1SccSourceAddress)&testData;
+    const asn1SccSourceAddress actualDataAddress =
+        (const asn1SccSourceAddress)&actualData;
+
+    /// Test if exclusive reading works correctly
+    controller_RI_RawMemoryAccess_ExclusiveReadWord_Pi(
+        &testDataAddress, &WHOLE_REGISTER_MASK, &actualData);
+    VALUE_EQUAL(actualData, EXPECTED_DATA_2);
+
+    /// Test if exclusive write works correctly
+    asn1SccByte writeStatus;
+    controller_RI_RawMemoryAccess_ExclusiveWriteWord_Pi(
+        &testDataAddress, &WHOLE_REGISTER_MASK, &EXPECTED_DATA_1, &writeStatus);
+    VALUE_EQUAL(testData, EXPECTED_DATA_1);
+    VALUE_EQUAL(writeStatus, EXPECTED_STATUS_SUCCESS);
+}
+
+static inline void Init_read_modify_write_failure_test()
+{
+    const asn1SccSourceAddress testDataAddress =
+        (const asn1SccSourceAddress)&testData;
+    asn1SccWord actualData;
+    controller_RI_RawMemoryAccess_ExclusiveReadWord_Pi(
+        &testDataAddress, &WHOLE_REGISTER_MASK, &actualData);
+    /// Any kind of interrupt needs to be triggered for exclusive read modify write to fail
+    Spi_enable_tx_interrupt();
+}
+
+/// This function tests the fulfillment of the requirement ETB_FUN-5030
```

Comment:
This trace should be added in the python test harness. Please look-up the example:
##
# \brief Generic_Linux_Multiple_Interfaces
# \SRS ETB-FUN-7010
# \SRS ETB-FUN-7020
@pytest.mark.parametrize('taste_project',
                         ['Generic_Linux_Multiple_Interfaces'],
                         indirect=True)
def test_generic_linux_multiple_interfaces(taste_project):